accounts-db/src/tiered_storage.rs

```diff
@@ -109,29 +115,36 @@ impl TieredStorage {
         skip: usize,
         format: &TieredStorageFormat,
     ) -> TieredStorageResult<Vec<StoredAccountInfo>> {
-        if self.is_read_only() {
-            return Err(TieredStorageError::AttemptToUpdateReadOnly(
-                self.path.to_path_buf(),
-            ));
-        }
-
-        if format == &HOT_FORMAT {
-            let result = {
-                let writer = HotStorageWriter::new(&self.path)?;
-                writer.write_accounts(accounts, skip)
-            };
-
-            // panic here if self.reader.get() is not None as self.reader can only be
-            // None since we have passed `is_read_only()` check previously, indicating
-            // self.reader is not yet set.
-            self.reader
-                .set(TieredStorageReader::new_from_path(&self.path)?)
-                .unwrap();
+        let was_written =
+            self.already_written
+                .compare_exchange(false, true, Ordering::AcqRel, Ordering::Relaxed);
```

Comment:
> However, if we expect only a single call to write_accounts(), then we can use swap instead of compare_exchange, and panic!() if was_written is ever true.

Yep, let's do it as it is the current behavior.  Thread-safety here is just to offer an extra guard.

Reply:
Updated to use `swap`.